### ISSUE TITLE
`eth_getLogs` returns different number of logs for different ranges

### ISSUE THREAD

**Issue by pawanjay176** [NethermindEth/nethermind#1980]
I am running a Lighthouse eth2 beacon node with a Nethermind client as my eth1 node on the Witti eth2 multiclient testnet which runs on the Goerli testnet (Contract address 0x42cc0FcEB02015F145105Cf6f19F90e9BEa76558). 

The deposit contract emits a log every time someone sends a deposit. We use these logs to construct the deposit merkle tree. For the purposes of constructing the deposit tree, it's important that we get all the deposit logs without any missing logs in between.

However, it seems like Nethermind gives different results when queried with different ranges using the `eth_getLogs` query.

On querying with 10 separate queries for the 1000 block range, I get the following
```
Found 0 logs in range 2777066 to 2777166
Found 0 logs in range 2777166 to 2777266
Found 0 logs in range 2777266 to 2777366
Found 1 logs in range 2777366 to 2777466
Found 0 logs in range 2777466 to 2777566
Found 0 logs in range 2777566 to 2777666
Found 2 logs in range 2777666 to 2777766
Found 0 logs in range 2777766 to 2777866
Found 6 logs in range 2777866 to 2777966
Found 0 logs in range 2777966 to 2778066
```
On running a single query for the entire 1000 block range, I get 
```
Found 7 logs in range 2777066 to 2778066
```
Ran into this issue on a `Nethermind/v1.8.41-0-8452b7da9-20200529/X64-Linux 5.4.0-33-generic/Core3.1.3`.

Note, that I don't always hit this issue. Synced another nethermind node and the 1000 range query returned 9 as it should.

**Comment by tkstanczak** [comment]
This is being looked at, we were able to reproduce it and tested the first fixes and need a bit more time.

**Comment by LukaszRozmej** [comment]
Hi @pawanjay176 great catch! There was in some rare circumstances one reused buffer leaking to a cache were it could corrupt high level bloom indexes we create for block ranges in order to speed up get_logs for wide range searches. This issue had extremely low chance of occurring on chain with dense blooms like mainnet. But only low chance on chain with sparse blooms like goerli.

Managed to fix it and will go in v1.8.44 and later.

We recommend resync. But if you want to keep DB, you can just remove 'bloom' folder in DB and put "Migrate":true under "Bloom" part of the config (see https://nethermind.readthedocs.io/en/latest/configuration.html#bloomconfig ).
